sql/atxs/atxs.go

```diff
@@ -569,27 +670,46 @@ func IterateAtxsData(
 		weight uint64,
 		base uint64,
 		height uint64,
+		nonce *types.VRFPostIndex,
+		isMalicious bool,
 	) bool,
 ) error {
 	_, err := db.Exec(
-		`select id, pubkey, epoch, coinbase, effective_num_units, base_tick_height, tick_count
-		from atxs where epoch between ?1 and ?2;`,
-		func(stmt *sql.Statement) {
-			stmt.BindInt64(1, int64(from.Uint32()))
-			stmt.BindInt64(2, int64(to.Uint32()))
-		},
+		`select
+                   a.id, a.pubkey, a.epoch, a.coinbase, a.effective_num_units,
+                   a.base_tick_height, a.tick_count, a.nonce,
+                   iif(idn.proof is null, 0, 1) as is_malicious
+		from atxs a left join identities idn on a.pubkey = idn.pubkey`,
+		// SQLite happens to process the query much faster if we don't
+		// filter it by epoch
+		// where a.epoch between ? and ?`,
+		// func(stmt *sql.Statement) {
+		// 	stmt.BindInt64(1, int64(from.Uint32()))
+		// 	stmt.BindInt64(2, int64(to.Uint32()))
+		// },
+		nil,
 		func(stmt *sql.Statement) bool {
+			epoch := types.EpochID(uint32(stmt.ColumnInt64(2)))
+			if epoch < from || epoch > to {
+				return true
+			}
 			var id types.ATXID
 			stmt.ColumnBytes(0, id[:])
 			var node types.NodeID
 			stmt.ColumnBytes(1, node[:])
-			epoch := types.EpochID(uint32(stmt.ColumnInt64(2)))
 			var coinbase types.Address
 			stmt.ColumnBytes(3, coinbase[:])
 			effectiveUnits := uint64(stmt.ColumnInt64(4))
 			baseHeight := uint64(stmt.ColumnInt64(5))
 			ticks := uint64(stmt.ColumnInt64(6))
-			return fn(id, node, epoch, coinbase, effectiveUnits*ticks, baseHeight, baseHeight+ticks)
+			var vrfNonce *types.VRFPostIndex
+			if stmt.ColumnType(7) != sqlite.SQLITE_NULL {
+				nonce := types.VRFPostIndex(stmt.ColumnInt64(7))
+				vrfNonce = &nonce
+			}
```

Comment:
I thought this column will never be NULL?

Reply:
As `NOT NULL` rule cannot be enforced yet due to test issues (see above), this makes it easier to check if it was violated (during warmup)